osquery/events/linux/auditdnetlink.cpp

```diff
@@ -392,54 +393,38 @@ bool AuditdNetlinkReader::configureAuditService() noexcept {
     }
   }
 
+  audit_rule_data rule = {};
+
   // Attempt to add each one of the rules we collected
   for (int syscall_number : monitored_syscall_list_) {
-    audit_rule_data rule = {};
     audit_rule_syscall_data(&rule, syscall_number);
-
-    // clang-format off
-    int rule_add_error = audit_add_rule_data(audit_netlink_handle_, &rule,
-      // We want to be notified when we exit from the syscall
-      AUDIT_FILTER_EXIT,
-
-      // Always audit this syscall event
-      AUDIT_ALWAYS
-    );
-    // clang-format on
-
-    // When exiting, don't remove the rules that were already installed, unless
-    // we have been asked to
-    if (rule_add_error >= 0) {
-      if (FLAGS_audit_debug) {
-        VLOG(1) << "Audit rule installed for syscall " << syscall_number;
-      }
-
-      installed_rule_list_.push_back(rule);
-      continue;
-    }
-
     if (FLAGS_audit_debug) {
-      VLOG(1) << "Audit rule for syscall " << syscall_number
-              << " could not be installed: " << (-errno);
+      VLOG(1) << "Audit rule queued for syscall " << syscall_number;
     }
+  }
 
-    if (FLAGS_audit_force_unconfigure) {
-      installed_rule_list_.push_back(rule);
-    }
+  // clang-format off
+  int rule_add_error = audit_add_rule_data(audit_netlink_handle_, &rule,
```

Comment:
The code should warn, possibly with `LOG(ERROR)`, when an error occurs. Since the rules are now grouped together, the error message could be something like `"Failed to install one or more Audit rules. Audit-based tables may not function as expected"`

Reply:
Thanks for taking look at the PR. I have updated the error log with `VLOG(ERROR)` which mentions that installing the rule has failed and with the resolved error code as well using `audit_errno_to_name()`.